autotest/gcore/vsicurl.py

```diff
@@ -507,6 +507,29 @@ def test_vsicurl_test_retry():
         assert '429' in error_msg
 
 
+def test_vsicurl_test_range_retry():
+
+    if gdaltest.webserver_port == 0:
+        pytest.skip()
+
+    handler = webserver.SequentialHandler()
+    handler.add('GET', '/test_retry/', 404)
+    handler.add('HEAD', '/test_retry/test.txt', 200, {'Content-Length': '6'})
+    handler.add('GET', '/test_retry/test.txt', 502, {'Content-Range': 'bytes 3-5/6'})
```

Comment:
due to the chunk caching logic which tries to 16384 bytes at once, I bet the Content-Range would be 'bytes 0-5/6'